android/app/src/main/kotlin/com/jabook/app/jabook/audio/MediaSessionManager.kt

```
@@ -109,10 +109,17 @@ public class MediaSessionManager(
-                // This happens when user clicks button in Quick Settings, notification, or lockscreen
-                if (reason == Player.PLAY_WHEN_READY_CHANGE_REASON_USER_REQUEST) {
+                // User commands and involuntary playback pauses must both close an active
+                // listening session. Other non-user transitions are informational only.
+                if (reason == Player.PLAY_WHEN_READY_CHANGE_REASON_USER_REQUEST ||
+                    (
+                        !playWhenReady &&
+                            (
+                                reason == Player.PLAY_WHEN_READY_CHANGE_REASON_AUDIO_FOCUS_LOSS ||
+                                    reason == Player.PLAY_WHEN_READY_CHANGE_REASON_AUDIO_BECOMING_NOISY
+                            )
+                    )
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)